### DISCUSSION TITLE
(Suggestion) Remove hard-coded bookmarks in Nemo, & let users fully customize what bookmars they see

### DISCUSSION THREAD

**Discussion by dabunginator** [linuxmint/discussions#583] [Ideas]
In Nemo, on the Places sidebar, bookmarks like Desktop can't be removed or hidden by users. Personally, I don't use my desktop often, so I really don't need a permanent bookmark for it. Other users may want to rearrange the hard-coded bookmarks to fit their needs, but those bookmarks can't be repositioned.

Along with removing hard-coded bookmarks, letting users add or remove sections would be nice too: sections like "My Computer".

**Comment by Jopp-gh** [comment]
Time ago, I suggested to upgrade the sidebar and the head part of Nemo:
https://github.com/orgs/linuxmint/discussions/425#discussioncomment-9531281

the vertical alignment of "bookmarks" eats just a lot of space for a few strings. A dynamic cloud tag of color labeled "bookmarks" would be much more useful and quick to navigate paths.

**Reply by anaximeno** [reply]
The bookmarks section is an interesting idea I wonder if it would be possible to implement it as an extension for Nemo https://github.com/linuxmint/nemo-extensions

**Reply by Jopp-gh** [reply]
Already possible with code libs like:
- https://github.com/wting/autojump
- https://github.com/rupa/z?tab=readme-ov-file

We are "just" missing a wrapper GUI to display a: 
- search mask to filter visible bookmarks,  
- coloured borders based on criteria like, user folder (no particular color needed) or sys folder (gray) or self added folder (accent color) or dynamically added folders, often visited locations (orange)
- some tension dynamic between labels to animate their collisions when sorting
- some sort logic to place most visited paths (bookmarks) on top of the tag cloud
- ability to drag-drop a new folder into this GUI

Yes, a plugin shouldn't be too hard to assemble for an experienced Linux dev. Unfortunately I myself have zero experience about gtk and python.  
One thing is to build code another thing is to know the right libs to call and write a concise code, so future updates are easy to update. Also, code needs to be performant, light and well embedded into the OS.
